freqtrade/data/entryexitanalysis.py

```diff
@@ -152,25 +152,44 @@ def _do_group_table_output(bigdf, glist):
                 logger.warning("Invalid group mask specified.")
 
 
+def _select_rows_within_dates(df, date_start=None, date_end=None):
+    if (date_start is not None):
+        df = df.loc[(df['date'] >= date_start)]
+
+    if (date_end is not None):
+        df = df.loc[(df['date'] < date_end)]
+
+    return df
+
+
+def _select_rows_by_entry_exit_tags(df, enter_reason_list, exit_reason_list):
+    if enter_reason_list and "all" not in enter_reason_list:
+        df = df.loc[(df['enter_reason'].isin(enter_reason_list))]
+
+    if exit_reason_list and "all" not in exit_reason_list:
+        df = df.loc[(df['exit_reason'].isin(exit_reason_list))]
+
+    return df
+
+
 def _print_results(analysed_trades, stratname, analysis_groups,
                    enter_reason_list, exit_reason_list,
-                   indicator_list, columns=None):
+                   indicator_list, columns=None,
+                   date_start=None, date_end=None):
```

Comment:
I don't quite see how this will satisfy the first point of "a user wants to select a subset"...

it's a private method which is not exposed to the user (and should not be used directly).
maybe you forgot to commit the parts where the user can provide his selection via `freqtrade backtesting-analysis` ?

Reply:
Yup this was the first step for the notebook update (so i can use the actual FT code instead of my buy_reasons script still). I'll add the user selection in the next PR.

Reply:
honestly? for now, i don't see why we need this as is.
it's adding parameters to an internal method (it starts with a `_`) - which are not used.

Please add the user interface part here, too.

also, if you consider this as a public facing method - then the name of the method is clearly wrong.